contracts/src/matching/HyperdriveMatchingEngineV2.sol

```diff
@@ -0,0 +1,659 @@
+        bytes32 _order2Hash
+    ) internal view returns (
+        uint256 bondMatchAmount
+    ) {
+        uint256 order1BondAmountUsed = orderBondAmountUsed[_order1Hash];
+        uint256 order2BondAmountUsed = orderBondAmountUsed[_order2Hash];
+
+        if (order1BondAmountUsed >= _order1.bondAmount || order2BondAmountUsed >= _order2.bondAmount) {
+            revert NoBondMatchAmount();
+        }
+
+        uint256 _order1BondAmount = _order1.bondAmount - order1BondAmountUsed;
+        uint256 _order2BondAmount = _order2.bondAmount - order2BondAmountUsed;
+            
+        bondMatchAmount = _order1BondAmount.min(_order2BondAmount);
+    }
+
+    /// @dev Handles the minting of matching positions.
+    /// @param _longOrder The order for opening a long position.
+    /// @param _shortOrder The order for opening a short position.
+    /// @param _baseTokenAmountLongOrder The amount of base tokens from the long 
+    ///        order.
+    /// @param _baseTokenAmountShortOrder The amount of base tokens from the short 
+    ///        order.
+    /// @param _cost The total cost of the operation.
+    /// @param _bondMatchAmount The amount of bonds to mint.
+    /// @param _baseToken The base token being used.
+    /// @param _hyperdrive The Hyperdrive contract instance.
+    /// @return The amount of bonds minted.
+    function _handleMint(
+        OrderIntent calldata _longOrder,
+        OrderIntent calldata _shortOrder,
+        uint256 _baseTokenAmountLongOrder,
+        uint256 _baseTokenAmountShortOrder,
+        uint256 _cost,
+        uint256 _bondMatchAmount,
+        ERC20 _baseToken,
+        IHyperdrive _hyperdrive
+    ) internal returns (uint256) {
+        // Transfer base tokens from long trader
+        _baseToken.safeTransferFrom(
+            _longOrder.trader,
+            address(this),
+            _baseTokenAmountLongOrder
+        );
+
+        // Transfer base tokens from short trader
+        _baseToken.safeTransferFrom(
+            _shortOrder.trader,
+            address(this),
+            _baseTokenAmountShortOrder
+        );
+
+        // Approve Hyperdrive
+        // @dev Use balanceOf to get the total amount of base tokens instead of 
+        //      summing up the two amounts, in order to open the door for poential 
+        //      donation to help match orders.
+        uint256 totalBaseTokenAmount = _baseToken.balanceOf(address(this));
+        uint256 baseTokenAmountToUse = _cost + TOKEN_AMOUNT_BUFFER;
+        if (totalBaseTokenAmount < baseTokenAmountToUse) {
+            revert InsufficientFunding();
+        }
+        _baseToken.forceApprove(address(_hyperdrive), baseTokenAmountToUse);
+
+        // Create PairOptions
+        IHyperdrive.PairOptions memory pairOptions = IHyperdrive.PairOptions({
+            longDestination: _longOrder.options.destination,
+            shortDestination: _shortOrder.options.destination,
+            asBase: true,
+            extraData: ""
```

Comment:
We need to add support for this. I'm not exactly sure what the best way to do this is, so let's just leave a TODO for now.